pkg/primaza/workercluster/agentapp.go

```diff
@@ -0,0 +1,225 @@
+	}
+
+	dep := obj.(*appsv1.Deployment)
+	r, err := cli.AppsV1().Deployments(namespace).Create(ctx, dep, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error creating deployment: %w", err)
+	}
+
+	return r, nil
+}
+
+func deleteAgentAppService(ctx context.Context, cli *kubernetes.Clientset, namespace string) error {
+	s := runtime.NewScheme()
+	if err := corev1.AddToScheme(s); err != nil {
+		return fmt.Errorf("decoder error: %w", err)
+	}
+	decode := serializer.NewCodecFactory(s).UniversalDeserializer().Decode
+
+	obj, _, err := decode([]byte(agentAppService), nil, nil)
+	if err != nil {
+		return fmt.Errorf("decoder error: %w", err)
+	}
+
+	svc := obj.(*corev1.Service)
+	if err := cli.CoreV1().Services(namespace).Delete(ctx, svc.Name, metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("error deleting service: %w", err)
+	}
+
+	return nil
+}
+
+func deleteAgentAppDeployment(ctx context.Context, cli *kubernetes.Clientset, namespace string) error {
+	s := runtime.NewScheme()
+	if err := appsv1.AddToScheme(s); err != nil {
+		return fmt.Errorf("decoder error: %w", err)
+	}
+
+	decode := serializer.NewCodecFactory(s).UniversalDeserializer().Decode
+
+	obj, _, err := decode([]byte(agentAppDeployment), nil, nil)
+	if err != nil {
+		return fmt.Errorf("decoder error: %w", err)
+	}
+
+	dep := obj.(*appsv1.Deployment)
+	if err := cli.AppsV1().Deployments(namespace).Delete(ctx, dep.Name, metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("error deleting deployment: %w", err)
+	}
+
+	return nil
+}
+
+const agentAppService string = `
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    control-plane: primaza-controller-agentapp
+    app.kubernetes.io/name: service
+    app.kubernetes.io/instance: agentapp-metrics-service
+    app.kubernetes.io/component: kube-rbac-proxy
+    app.kubernetes.io/created-by: primaza
+    app.kubernetes.io/part-of: primaza
+  name: agentapp-metrics-service
+spec:
+  ports:
+  - name: https
+    port: 8443
+    protocol: TCP
+    targetPort: https
+  selector:
+    control-plane: primaza-controller-agentapp
+`
+
+const agentAppDeployment string = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: primaza-controller-agentapp
+  labels:
+    control-plane: primaza-controller-agentapp
+    app.kubernetes.io/name: deployment
+    app.kubernetes.io/instance: primaza-controller-agentapp
+    app.kubernetes.io/component: agentapp-manager
+    app.kubernetes.io/created-by: primaza
+    app.kubernetes.io/part-of: primaza
+spec:
+  selector:
+    matchLabels:
+      control-plane: primaza-controller-agentapp
+  replicas: 1
+  template:
+    metadata:
+      annotations:
+        kubectl.kubernetes.io/default-container: manager
+      labels:
+        control-plane: primaza-controller-agentapp
+    spec:
+      securityContext:
+        runAsNonRoot: true
+      containers:
+      - command:
+        - /manager
+        args:
+        - --leader-elect
+        image: agentapp:latest
```

Comment:
hardcoded? is this temporary? Lets talk about the name of these images in scrum or refinement.

Reply:
yes, that's a placeholder for now and will be replaced in next PRs